<?xml version="1.0"?>

<!-- Launch file for the starting mission for the thesis of Boris Dehem-->
<!-- Work In Progress -->
<launch>
  <arg name="threshold" default="123.4" />

  <!-- IPv4 address of your drones -->
  <arg name="drone_name" default="arnold" />
  <arg name="mb1_ip" default="192.168.1.1" />
  <!-- Ultrasound frequency (7 or 8). -->
  <arg name="freq" default="8" />


  <node name="vision_gui" pkg="boris_drone" type="vision_gui" output="screen">
    <param name="draw_keypoints" value="true" />
    <param name="draw_target" value="true" />
  </node>

  <node name="boris_drone_manual_pose_estimation" pkg="boris_drone" type="manual_pose_estimation" output="screen">
  </node>

  <!-- front camera -->
  <rosparam param="cam_matrix">[529.1, 0.0, 350.6, 0.0, 529.1, 182.2, 0.0, 0.0, 1.0]</rosparam>
  <rosparam param="img_size">[735.0, 360.0]</rosparam>
  <!-- front camera -->


  <node name="boris_drone_mapping_node" pkg="boris_drone" type="mapping_node" output="screen">
    <param name="2D_noise" value="false" />
    <param name="3D_noise" value="false" />
    <param name="threshold_kf_match" value="$(arg threshold)" />
  </node>

  <node name="boris_drone_bundle_adjuster" pkg="boris_drone" type="bundle_adjuster" output="screen">
  </node>

  <node pkg="rosbag" type="play" name="player" output="screen" args="/home/bor/bagfiles/benchmark_precision.bag"/>

  <node pkg="rosbag" type="record" name="recorder" output="screen" args="-O /home/bor/bagfiles/benchmark_point_threshold/thresh_$(arg threshold) /pose_visual /pose_estimation /benchmark"/>

</launch>
